WalletWasabi/WabiSabi/Client/AliceClient.cs

```diff
@@ -50,10 +53,25 @@ public async Task RegisterInputAsync(CancellationToken cancellationToken)
 
 		public async Task ConfirmConnectionAsync(TimeSpan connectionConfirmationTimeout, IEnumerable<long> amountsToRequest, IEnumerable<long> vsizesToRequest, CancellationToken cancellationToken)
 		{
-			while (!await TryConfirmConnectionAsync(amountsToRequest, vsizesToRequest, cancellationToken).ConfigureAwait(false))
+			do
 			{
-				await Task.Delay(connectionConfirmationTimeout / 2, cancellationToken).ConfigureAwait(false);
+				using CancellationTokenSource timeout = new(connectionConfirmationTimeout / 2);
+				using CancellationTokenSource cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, timeout.Token);
+
+				try
+				{
+					await RoundStatusUpdater
+						.CreateRoundAwaiter(
+							RoundId,
+							roundState => roundState.Phase == Phase.ConnectionConfirmation,
+							cts.Token)
+						.ConfigureAwait(false);
+				}
+				catch (OperationCanceledException)
```

Comment:
In case the `OperationCanceledException.CancellationToken == cancellationToken) then we should let it bubble. Something similar can be achieved (but it is worse btw)  by simply calling `cancellationToken.ThrowIfCancellationRequested();` inside the catch block.

Reply:
ACK